core/src/main/java/org/elasticsearch/index/analysis/CustomAnalyzerProvider.java

```diff
@@ -77,13 +78,27 @@ public void build(AnalysisService analysisService) {
             tokenFilters.add(tokenFilter);
         }
 
-        int positionOffsetGap = analyzerSettings.getAsInt("position_offset_gap", 0);
-        int offsetGap = analyzerSettings.getAsInt("offset_gap", -1);
+        int positionIncrementGap = 0;
+        if (analyzerSettings.getAsMap().containsKey("position_offset_gap")){
+            if (Version.indexCreated(indexSettings).before(Version.V_2_0_0)){
+                if (analyzerSettings.getAsMap().containsKey("position_increment_gap")){
+                    throw new IllegalArgumentException("Option [position_offset_gap] in custom Analyzer [" + name() +
```

Comment:
I don't think this is the right error message in this case.

I really think we'd be better off with something like:
```java
int positionIncrementGap = 0;
if (analyzerSettings.getAsMap().containsKey("position_offset_gap") {
   log some deprecation notice
   positionIncrementGap = analyzerSettings.getAsInt("position_offset_gap", positionIncrementGap);
}
positionIncrementGap = analyzerSettings.getAsInt("position_increment_gap", positionIncrementGap);
```

Then we have a consistent story in the migration documentation:
`position_offset_gap` is deprecated. Use `position_increment_gap` instead. If you use both `position_increment_gap` will win